### DISCUSSION TITLE
Parametrise source files so vd command logs can be replayed with different sources

### DISCUSSION THREAD

**Discussion by ghost** [saulpw/visidata#1187] [Ideas]
Perhaps this is already a feature...

Suppose you have two sets of files:

A1, A2, A3
B1, B2, B3

You want to open A1 and B1, join them, do some manipulations, create/delete/rename columns, filter, etc.
You then want to apply the exact same set of operations to A* and B*.

Is this possible? Or would it make sense to add a way parametrise command logs so this was possible?

e.g. something like:
```
vd --play fancy.vd --sources A=A2.csv B=B2.csv
```

Then the command log would refer to `A` and `B` as the sources, rather than `A2.csv` and `B2.csv`.
Essentially this would be introducing variables to command logs.

I tried just opening the command log in vim and doing a search and replace, but my command log wasn't robust enough to handle this! (I probably did something that didn't translate to the other file)

**Comment by geekscrapy** [comment]
I believe that if you delete the input filename in the .vd it works with whatever file you provide on the cli.

In addition, if the .vd doesn't have a sheet name it will work with the top sheet.

Using these two features you should be able to run the commands from 1 sheet, on another 👍

**Reply by saulpw** [reply]
Maybe we should make it so that the cmdlog doesn't include the sheet column by default, but only if the sheet is different between commands. Then at least single-sheet workflows can be replayed on new files without hassle.

**Reply by geekscrapy** [reply]
Yea, I think this is logical. My guess is that most of the batch processing is not too complicated and generally uses one sheet, and something you don't want to spend a ton of time doing. As opposed to more complicated processing where it takes a bit of debugging etc. and sometimes you may as well write a separate script?

**Reply by ghost** [reply]
Thank you! I'll give it a try. My particular use-case involved the following steps:

- Open two files
- Create a numeric index in the first file starting at 2 (`i` `=A+1`)
- Inner join the first file to the second file on the numeric index (second file already had an index)
- Filter a specific column (`Shift+F` `/search` `Enter`)
- Save output to new file

Is this likely to work on another pair of files with the same structure?
